rules/TypeDeclaration/Rector/ClassMethod/ParamAnnotationIncorrectNullableRector.php

```diff
@@ -0,0 +1,197 @@
+        $wasAnyParamTagUpdated = $this->wasUpdateOfParamTagsRequired($phpDocNode, $node, $phpDocInfo);
+
+        if (! $wasAnyParamTagUpdated) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    private function matchParamByName(string $desiredParamName, ClassMethod|Function_ $node): ?Param
+    {
+        foreach ($node->getParams() as $param) {
+            $paramName = $this->nodeNameResolver->getName($param);
+            if ('$' . $paramName !== $desiredParamName) {
+                continue;
+            }
+
+            return $param;
+        }
+
+        return null;
+    }
+
+    private function changeParamType(PhpDocInfo $phpDocInfo, Type $newType, Param $param, string $paramName): void
+    {
+        // better skip, could crash hard
+        if ($phpDocInfo->hasInvalidTag('@param')) {
+            return;
+        }
+
+        $typeNode = $this->staticTypeMapper->mapPHPStanTypeToPHPStanPhpDocTypeNode($newType, TypeKind::PARAM());
+        $paramTagValueNode = $phpDocInfo->getParamTagValueByName($paramName);
+        // override existing type
+        if ($paramTagValueNode !== null) {
+            // already set
+            $currentType = $this->staticTypeMapper->mapPHPStanPhpDocTypeNodeToPHPStanType(
+                $paramTagValueNode->type,
+                $param
+            );
+            if ($this->typeComparator->areTypesEqual($currentType, $newType)) {
+                return;
+            }
+
+            $paramTagValueNode->type = $typeNode;
+        } else {
+            $paramTagValueNode = $this->paramPhpDocNodeFactory->create($typeNode, $param);
+            $phpDocInfo->addTagValueNode($paramTagValueNode);
+        }
+    }
+
+    private function wasUpdateOfParamTagsRequired(
+        PhpDocNode $phpDocNode,
+        ClassMethod|Function_ $node,
+        PhpDocInfo $phpDocInfo
+    ): bool {
+        $paramTagValueNodes = $phpDocNode->getParamTagValues();
+        $paramTagWasUpdated = false;
+        foreach ($paramTagValueNodes as $paramTagValueNode) {
+            if ($paramTagValueNode->type === null) {
+                continue;
+            }
+
+            $param = $this->matchParamByName($paramTagValueNode->parameterName, $node);
+            if (! $param instanceof Param) {
+                continue;
+            }
+
+            $docType = $this->staticTypeMapper->mapPHPStanPhpDocTypeNodeToPHPStanType(
+                $paramTagValueNode->type,
+                $node
+            );
+            $updatedPhpDocType = $this->phpDocNullableTypeHelper->resolveUpdatedPhpDocTypeFromPhpDocTypeAndParamNode(
+                $docType,
+                $param
+            );
+
+            if (! $updatedPhpDocType instanceof Type) {
+                continue;
+            }
+
+            $this->changeParamType($phpDocInfo, $updatedPhpDocType, $param, $paramTagValueNode->parameterName);
+            $paramTagWasUpdated = true;
+        }
+
+        return $paramTagWasUpdated;
```

Comment:
$paramTagWasUpdated can be removed and directly return:

```suggestion
        return $phpDocInfo->hasChanged();
```